app/graphql/types/language_mapping_type.rb

```diff
@@ -4,7 +4,7 @@
   name 'LanguageMapping'
   description 'A mapping between two languages'
 
-  field :id, !types.ID do
+  field :id, !types.Int do
```

Comment:
Why this change?

Reply:
It's because language mappings don't have a slug. The `ID` was a bug. We query by the numerical `id` column.

Reply:
It might still make sense to use ID here. AFAIK Apollo uses ID fields as primary keys for caching.

Reply:
I think it's not an `ID` typed field, but an `id` named field:

> By default, InMemoryCache will attempt to use the commonly found primary keys of id and _id for the unique identifier if they exist along with __typename on an object.

https://www.apollographql.com/docs/react/basics/caching.html#normalization


This could mean that we should rename `locId` fields in the queries to `id`...

Reply:
... but it seems we don't need to do that. See the next paragraphs of the above link:

> This “getter” behavior for unique identifiers can be configured manually via the dataIdFromObject option passed to the InMemoryCache constructor, so you can pick which field is used if some of your data follows unorthodox primary key conventions so it could be referenced by any query.
>
> For example, if you wanted to key off of the key field for all of your data, you could configure dataIdFromObject like so:
>
> ```
> const cache = new InMemoryCache({
>   dataIdFromObject: object => object.key || null
> });
> ```
> Note that Apollo Client doesn’t add the type name to the cache key when you specify a custom dataIdFromObject, so if your IDs are not unique across all objects, you might want to include the __typename in your dataIdFromObject.